### PR TITLE
fix(module select): simulate setTimeout, ensure that the click event is processed properly

### PR DESCRIPTION
**Pull request by zxyao145**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#3521

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    fix: simulate setTimeout, ensure that the click event is processed properly |
| 🇨🇳 Chinese |     模拟 setTimeout，确保 click 事件处理完毕      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/core/Component/Overlay/OverlayTrigger.razor.cs [hunk 1]
```diff
@@ -479,8 +479,13 @@ protected virtual async Task OnTriggerContextmenu(MouseEventArgs args)
...
         }
 
-        protected virtual void OnMouseUp(JsonElement element)
+        /// <summary>
+        /// on document mouse up
+        /// </summary>
+        /// <param name="element"></param>
+        protected virtual async void OnMouseUp(JsonElement element)
         {
+            await Task.Delay(4);
```
Comment (ElderJames) [inline components/core/Component/Overlay/OverlayTrigger.razor.cs:488]: Why is it 4ms?

Comment (zxyao145) [reply components/core/Component/Overlay/OverlayTrigger.razor.cs:488]: In JS, 4ms is a better configuration, and I think the same goes for Blazor.

https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-settimeout

**Comment by ElderJames** [comment]
> Meet another issue that cause exception after open two filter. table-filter-forcus-issue [ table-filter-forcus-issue ](https://private-user-images.githubusercontent.com/7550366/291645272-f33d4833-17ef-4b0e-836a-c158dae0e141.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNjk0MTAsIm5iZiI6MTcwMzE2OTExMCwicGF0aCI6Ii83NTUwMzY2LzI5MTY0NTI3Mi1mMzNkNDgzMy0xN2VmLTRiMGUtODM2YS1jMTU4ZGFlMGUxNDEuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjFUMTQzMTUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzY1M2JlZjg1YTY5NzBhNjA4ZDI2NzJlNmRlMjc4NjI1OWFjY2I2Zjc1MDg1ZGVkNTQ1MTEzMjc5ZDc2NTI0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.VmqLrcEpxv3okKnJwWC5MakImZxic-oBQ_XUSTMC0hU) [ ](https://private-user-images.githubusercontent.com/7550366/291645272-f33d4833-17ef-4b0e-836a-c158dae0e141.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNjk0MTAsIm5iZiI6MTcwMzE2OTExMCwicGF0aCI6Ii83NTUwMzY2LzI5MTY0NTI3Mi1mMzNkNDgzMy0xN2VmLTRiMGUtODM2YS1jMTU4ZGFlMGUxNDEuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjFUMTQzMTUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzY1M2JlZjg1YTY5NzBhNjA4ZDI2NzJlNmRlMjc4NjI1OWFjY2I2Zjc1MDg1ZGVkNTQ1MTEzMjc5ZDc2NTI0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.VmqLrcEpxv3okKnJwWC5MakImZxic-oBQ_XUSTMC0hU)

This have been fixed via https://github.com/ant-design-blazor/ant-design-blazor/pull/3543

File: tests/AntDesign.Tests/Select/Renders/Select.Render.Tests.razor [hunk 2]
```diff
@@ -45,62 +45,6 @@
         arrow.Should().NotBeNull();
     }
 
-    [Fact] //covers scenario from issue #1989
-    public async Task Render_select_dropdown_on_arrow_click()
```
Comment (ElderJames) [inline tests/AntDesign.Tests/Select/Renders/Select.Render.Tests.razor:49]: Why do you remove so many tests?

Comment (zxyao145) [reply tests/AntDesign.Tests/Select/Renders/Select.Render.Tests.razor:49]: BUnit cannot simulate the client events bound through JavaScript.

Comment (zxyao145) [reply tests/AntDesign.Tests/Select/Renders/Select.Render.Tests.razor:49]: Inblazor, using `@onclick:stopPromotion="@true"` on the child component does not prevent the parent component from responding to click events. Therefore, I replaced the event binding in blazer with binding through Js. 

Due to the above reasons, so this test is no longer executable.

**Review by ElderJames — APPROVED** [review]

**Pull request merged**